taskcluster/app_services_taskgraph/build_config.py

```diff
@@ -0,0 +1,33 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from __future__ import absolute_import, print_function, unicode_literals
+
+import os
+import yaml
+
+from taskgraph.util.memoize import memoize
+
+
+def get_components():
+    build_config = _read_build_config()
+    return [{
+        'version': build_config['libraryVersion'],
+        'name': name,
+        'path': project['path'],
+        'artifactId': project['artifactId'],
+        'publications': [{
+            'name': publication['name'],
+            'type': publication['type'],
+        } for publication in project['publications']]
+    } for (name, project) in build_config['projects'].items()]
+
+
+@memoize
```

Comment:
Nit: Shouldn't we memoize `get_components` instead since it applies the same transformation every time?

Reply:
We definitely can. I guess this solution wasn't implemented because Fenix has slightly different constraints https://github.com/mozilla-mobile/fenix/blob/8be6c77695a7dd5d990999d7bcb5ec09119d3e93/taskcluster/fenix_taskgraph/gradle.py#L33

From what I understand, transforming the data in `get_components()` is not costly and we won't save much by moving `@memoize` there. 

We can address this in a followup, if needed.

Reply:
Added as follow-up.